src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -132,7 +133,6 @@ class AuthScreens extends React.Component {
         if (this.unsubscribeGroupShortcut) {
             this.unsubscribeGroupShortcut();
         }
-        Session.cleanupSession();
```

Comment:
Not following this change. Why should we only do these tasks when the user explicitly signs out? 

It seems like on sign out this series of events will happen:

- `authToken` will clear
- `AuthScreens` will unmount as there is no `authToken`
- Cleanup happens as a side effect of losing the `authToken`

Reply:
We're moving this out of `componentWillUnmount` because that also runs when you press the back button to leave the app on Android. Are there other flows where we should clear session? Like when the authToken expires?

Reply:
Just tried this [SO](https://stackoverflowteams.com/c/expensify/questions/7826), which gets the token to expire however NewDot doesn't really react (on this branch and `main`).

Reply:
Oh darn, to clarify, you're saying that the `AuthScreens` `componentWillUnmount()` runs when the back button is pressed to leave the app on Android?

I wouldn't mind looking at this for a second. Why does that happen? And does it happen on a release build or just dev? Does `componentDidMount()` get called again when we bring the app to the foreground?

My working understanding of stuff is that we can use the `AuthScreens` mounting and unmount lifecycles as a way to "set up" or "tear down" things that should always happen when the user logs in or out. That is 100% based on the assumption that `componentWillUnmount()` only fires when we previously had an `authToken` but lost it for some reason (manually signing out or manually clearing storage are the two cases I can think of).

I think this is kind of easier to reason about because it is very binary - you either have an `authToken` or you don't. And if you had one before, but don't have one now then that is a "user logged out" scenario.

`AuthScreens` unmounting for some other reason definitely creates a problem for that idea 😄 

>Are there other flows where we should clear session? Like when the authToken expires?

The `authToken` can expire, but under normal conditions it will get refreshed. A user could manually clear their `authToken` from storage and that would effectively be the same as signing out without explicitly calling `signOut()`.

Reply:
> Oh darn, to clarify, you're saying that the AuthScreens componentWillUnmount() runs when the back button is pressed to leave the app on Android?

Yup that's correct. Specifically when you're on the home screen, press back to leave the app. This kills the app and AuthScreens gets unmounted.

> I wouldn't mind looking at this for a second. Why does that happen?

I haven't looked too closely, but I think it's a quirk with `react-navigation`. Pressing the back button usually pops a screen off the stack so I'm guessing that also happens even when you press back to exit the app. Unmounting screens might also just be part of the teardown process when the app gets killed.

>  And does it happen on a release build or just dev?

Both. This issue has been breaking notifications for me (and many others probably) on the prod app. Eventually I was able repro the issue dev.

Reply:
Ok cool I think in that case we need to hook the "clean up" parts of the app an Onyx handler that detects that the `authToken` has been cleared or something? Otherwise, it seems there is code that won't run until we explicitly call `signOut()` and seems like the only place where we do that is here:

https://github.com/Expensify/App/blob/0fcebd329567040d8832a314205ab61e602488fb/src/pages/settings/InitialSettingsPage.js#L214-L222

Reply:
I think it would also be good to maybe mention somewhere in a code comment about the quirky android behavior. I feel like the code is in a slightly inconsistent state where:

- "setup" tasks mostly are find to happen when `AuthScreens` mounts
- "teardown" tasks should not happen when `AuthScreens` unmounts

And I think what we've really got is something like...

- things that need to happen when AuthScreens mounts/unmounts (e.g. subscribing to Pusher, calling OpenApp)
- things that need to happen when a user is signed in or signed out (e.g. de/registering for Push Notifications)

Reply:
Yeah I think using an Onyx handler could be a good alternative. We could use that specifically for push notifications subscribe/unsubscribe and leave the other session cleanup as it was. I'll draft that soon